src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductOptionFactory.php

```diff
@@ -0,0 +1,136 @@
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DataFixtures\Factory;
+
+use Sylius\Component\Core\Formatter\StringInflector;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\Component\Product\Model\ProductOption;
+use Sylius\Component\Product\Model\ProductOptionInterface;
+use Sylius\Component\Product\Model\ProductOptionValueInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Zenstruck\Foundry\ModelFactory;
+use Zenstruck\Foundry\Proxy;
+
+/**
+ * @extends ModelFactory<ProductOptionInterface>
+ *
+ * @method static ProductOptionInterface|Proxy createOne(array $attributes = [])
+ * @method static ProductOptionInterface[]|Proxy[] createMany(int $number, array|callable $attributes = [])
+ * @method static ProductOptionInterface|Proxy find(object|array|mixed $criteria)
+ * @method static ProductOptionInterface|Proxy findOrCreate(array $attributes)
+ * @method static ProductOptionInterface|Proxy first(string $sortedField = 'id')
+ * @method static ProductOptionInterface|Proxy last(string $sortedField = 'id')
+ * @method static ProductOptionInterface|Proxy random(array $attributes = [])
+ * @method static ProductOptionInterface|Proxy randomOrCreate(array $attributes = [])
+ * @method static ProductOptionInterface[]|Proxy[] all()
+ * @method static ProductOptionInterface[]|Proxy[] findBy(array $attributes)
+ * @method static ProductOptionInterface[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method static ProductOptionInterface[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method ProductOptionInterface|Proxy create(array|callable $attributes = [])
+ */
+final class ProductOptionFactory extends ModelFactory implements ProductOptionFactoryInterface
+{
+    public function __construct(
+        private FactoryInterface $productOptionFactory,
+        private FactoryInterface $productOptionValueFactory,
+        private RepositoryInterface $localeRepository,
+    ) {
+        parent::__construct();
+    }
+
+    public function withCode(string $code): self
+    {
+        return $this->addState(['code' => $code]);
+    }
+
+    public function withName(string $name): self
+    {
+        return $this->addState(['name' => $name]);
+    }
+
+    public function withValues(array $values): self
+    {
+        return $this->addState(['values' => $values]);
+    }
+
+    protected function getDefaults(): array
+    {
+        return [
+            'code' => null,
+            'name' => self::faker()->words(3, true),
+            'values' => null,
+        ];
+    }
+
+    protected function initialize(): self
+    {
+        return $this
+            ->instantiateWith(function(array $attributes): ProductOptionInterface {
+                $code = $attributes['code'] ?? StringInflector::nameToCode($attributes['name']);
+
+                $values = $attributes['values'] ?? null;
```

Comment:
This statement does nothing.
```suggestion
```

Reply:
the usage is just below.

Reply:
Yes, but it does nothing, because it translates to: if value is null, return null.
It's the same as writing `$values = $attributes['values']` or just using `$attributes['values']` directly.

Reply:
In other words: it's redundant.

Reply:
I tried something else.